src/imgui/ImGuiMachine.cc

```diff
@@ -142,6 +156,46 @@ void ImGuiMachine::paintSelectMachine(const MSXMotherBoard* motherBoard)
 			ImGui::Separator();
 		}
 
+		auto showMachine = [&](MachineInfo& info, bool doubleClickToSelect) {
+			bool ok = getTestResult(info).empty();
+			im::StyleColor(!ok, ImGuiCol_Text, getColor(imColor::ERROR), [&]{
+				bool selected = info.configName == newMachineConfig;
+				if (ImGui::Selectable(info.displayName.c_str(), selected,
+						doubleClickToSelect ? ImGuiSelectableFlags_AllowDoubleClick: ImGuiSelectableFlags_None)) {
+					newMachineConfig = info.configName;
+					if (ok && (doubleClickToSelect ? ImGui::IsMouseDoubleClicked(ImGuiMouseButton_Left) : true)) {
+						showSelectMachine = false; // close window
+						manager.executeDelayed(makeTclList("machine", newMachineConfig));
+						addRecentItem(recentMachines, newMachineConfig);
+					}
+				}
+				if (selected) {
+					if (ImGui::IsWindowAppearing()) ImGui::SetScrollHereY();
+				}
+				if (ImGui::IsItemHovered(ImGuiHoveredFlags_DelayNormal | ImGuiHoveredFlags_NoSharedDelay | ImGuiHoveredFlags_Stationary)) {
+					im::ItemTooltip([&]{
+						printConfigInfo(info);
+					});
+				}
+			});
+		};
+
+		im::TreeNode("Recently used", recentMachines.empty() ? ImGuiTreeNodeFlags_None : ImGuiTreeNodeFlags_DefaultOpen, [&]{
+			if (recentMachines.empty()) {
+				ImGui::TextUnformatted("(none)"sv);
+			} else {
+				MachineInfo* info = nullptr;
+				im::Combo("##recent", "Select recently used machine...", [&]{
```

Comment:
Detail: about the text "**Select** recently used machine ...". This not just selects a machine, it replaces the current machine and start the selected one. Maybe replace *Select* with *Switch to*?

Several of the buttons in this window have tool-tips explaining what will happen if you press them. Maybe we could add a similar tooltip here on the closed combo box?  (But we'd have to verify that it's possible to have tool-tips only on the combo-box itself, not on the drop-down list).

Reply:
OK, changed it in the latest version. Maybe you have a better idea on what the text should be though, because it's quite duplicate with the 'preview' of the Combo. But indeed, the tooltip works as you expect.

Reply:
`If you click on one of these recently used machines, you immediately switch to them.`
The "If you click on..." that part is redundant. And you don't say why *immediately* switching might be bad. So I propose something like this (the *replace* word is important):
`Replace the current machine with a recently used other machine.`
Or even just this (to remove the redundancy with the *preview* text).
`Replace the current machine.`
or
`Replace the current with the selected machine.`

Apart from this detail I think the PR is ready. So feel free to merge.

Reply:
OK, changed it to that last suggestion in the latest version.